include/drivers/adp536x.h

```diff
@@ -4,19 +4,182 @@
  * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
  */
 
+#ifndef ADP536X_H_
+#define ADP536X_H_
+
+/**@file adp536x.h
+ *
+ * @brief Driver for ADP536X.
+ * @defgroup adp536x
+ * @{
+ */
+
 #include <zephyr.h>
 
+#define ADP536X_VBUS_ILIM_50mA  0x00
+#define ADP536X_VBUS_ILIM_100mA 0x01
+#define ADP536X_VBUS_ILIM_150mA 0x02
+#define ADP536X_VBUS_ILIM_200mA 0x03
+#define ADP536X_VBUS_ILIM_250mA 0x04
+#define ADP536X_VBUS_ILIM_300mA 0x05
+#define ADP536X_VBUS_ILIM_400mA 0x06
+#define ADP536X_VBUS_ILIM_500mA 0x07
+
+#define ADP536X_CHRG_CURRENT_10mA       0x00
+#define ADP536X_CHRG_CURRENT_50mA       0x04
+#define ADP536X_CHRG_CURRENT_100mA      0x09
+#define ADP536X_CHRG_CURRENT_150mA      0x0E
+#define ADP536X_CHRG_CURRENT_200mA      0x13
+#define ADP536X_CHRG_CURRENT_250mA      0x18
+#define ADP536X_CHRG_CURRENT_300mA      0x1D
+#define ADP536X_CHRG_CURRENT_320mA      0x1F
+
+#define ADP536X_OC_CHRG_THRESHOLD_25mA  0x00
+#define ADP536X_OC_CHRG_THRESHOLD_50mA  0x01
+#define ADP536X_OC_CHRG_THRESHOLD_100mA 0x02
+#define ADP536X_OC_CHRG_THRESHOLD_150mA 0x03
+#define ADP536X_OC_CHRG_THRESHOLD_200mA 0x04
+#define ADP536X_OC_CHRG_THRESHOLD_250mA 0x05
+#define ADP536X_OC_CHRG_THRESHOLD_300mA 0x06
+#define ADP536X_OC_CHRG_THRESHOLD_400mA 0x07
+
+/**
+ * @brief Initialize ADP536X.
+ *
+ * Retrieves the device structure of the ADP536X device.
```

Comment:
I think this can be skipped or rephrased, as the device struct isn't made available to the user through this function

Reply:
I will skip this comment.